### PR TITLE
Ad/yet more enhancement

### PR DESCRIPTION
**Pull request by adonesky1**
- Makes the `connect` method asynchronous in the SDK and provider classes to handle race conditions and ensure proper connection before proceeding.
- Updates `MetaMaskMultichainProvider` to wait briefly after attempting to connect, allowing the `onDisconnect` event to fire if the connection fails, and checking the `isConnected` flag accordingly.
- Changes the `invokeMethodResults` data structure to store both the result and the request object. 
- Moves SDK exports to a new `index.ts` file 
- Adds a `README.md` for the SDK

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -206,27 +206,37 @@ function App() {
...
+                    (method as MethodObject).name === 'eth_blockNumber',
+                );
+
+                const defaultRequest = {
+                  method: 'wallet_invokeMethod',
+                  params: {
+                    scope,
+                    request: openRPCExampleToJSON(example as MethodObject),
+                  },
+                };
```
Comment (jiexi) [inline src/App.tsx:230]: do we need to store the wrapper? can we add it on later when we make the call in `handleInvokeMethod`? doesn't really matter though

Comment (adonesky1) [reply src/App.tsx]: yeah... ok if we make this enhancement later?

File: src/App.tsx [hunk 2]
```diff
@@ -740,9 +759,11 @@ function App() {
                           [scope]: newAddress,
                         }));
 
-                        // modify the method request if it's a signing method (inject selected address)
                         const currentMethod = selectedMethods[scope];
-                        if (currentMethod && currentMethod in SIGNING_METHODS) {
+                        if (
+                          currentMethod &&
+                          currentMethod in METHODS_REQUIRING_PARAM_INJECTION
+                        ) {
```
[the remark this answers is not in the document]
Comment (adonesky1) [reply src/App.tsx]: done here: https://github.com/MetaMask/test-dapp-multichain/pull/20/commits/4c41fa79b1f8de621400e518ee4f77640dd0a2f5

File: src/App.tsx (hunk 1 above)
Comment (jiexi) [reply src/App.tsx]: yeah

File: src/sdk/index.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import SDK from './SDK';
+import { useSDK } from './useSDK';
+
+export { SDK, useSDK };
```
Comment (jiexi) [inline src/sdk/index.ts:4]:
```
export SDK from './SDK';
export { useSDK } from './useSDK';
```

does this work? sorry i'm being annoying lol

File: src/sdk/README.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+      return;
+    }
+  } catch (error) {
+    console.error('Connection error:', error);
+    return;
+  }
+
+  // Create a session
+  try {
+    const scopes = ['eip155:1', 'eip155:137']; // Replace with desired chain IDs
```
Comment (jiexi) [inline src/sdk/README.md:128]:
```suggestion
    const scopes = ['eip155:1', 'eip155:1337']; // Replace with desired chain IDs
```

File: src/sdk/README.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+      console.log('Connected to MetaMask');
+    } catch (error) {
+      console.error('Connection error:', error);
+    }
+  };
+
+  // Create a session with specified chain IDs (e.g., Ethereum Mainnet and Polygon)
+  const handleCreateSession = async () => {
+    try {
+      const scopes = ['eip155:1', 'eip155:137'];
```
Comment (jiexi) [inline src/sdk/README.md:55]:
```suggestion
      const scopes = ['eip155:1', 'eip155:1337'];
```

File: src/sdk/index.ts (hunk 3 above)
Comment (adonesky1) [reply src/sdk/index.ts]: done here: https://github.com/MetaMask/test-dapp-multichain/pull/20/commits/c5102b213e10b729fcc2fb63164f1af1cb4f9277

File: src/sdk/README.md (hunk 4 above)
Comment (adonesky1) [reply src/sdk/README.md:128]: This is actually Polygon's chainId

File: src/sdk/README.md (hunk 5 above)
Comment (adonesky1) [reply src/sdk/README.md:55]: see above

**Review by jiexi — APPROVED** [review]

File: src/sdk/README.md (hunk 4 above)
Comment (jiexi) [reply src/sdk/README.md:128]: is that a pokemon?

**Pull request merged**